### PR TITLE
Add navigation container tests

### PR DESCRIPTION
**Pull request by joshuatf**
Adds tests around basic nav container interaction.

### Detailed test instructions:

1. On a new site, finish the store setup wizard, but don't hide the task list.
2. Navigate to a WooCommerce Admin Analytics page.
3. Note the menu is under the "Analytics" level.
4. Click the "Store Setup" link.
5. Note that the navigation level automatically is updated to the root level where the "Home" item is marked active.
6. Make sure tests pass.

### CODE REVIEW

File: TESTING-INSTRUCTIONS.md [hunk 1]
```diff
@@ -3,6 +3,31 @@ Testing instructions
 
 ## Unreleased
 
+### Add navigation container tests #6464
+
+1. On a new site, finish the store setup wizard, but don't hide the task list.
+2. Navigate to a WooCommerce Admin Analytics page.
+3. Note the menu is under the "Analytics" level.
+4. Click the "Store Setup" link.
```
Comment (psealock) [inline TESTING-INSTRUCTIONS.md:11]:
```suggestion
4. Click the "Store Setup" link in the top right hand corner.
```

File: client/navigation/components/container/test/index.js [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+			window.location = new URL(
+				getAdminLink( 'admin.php?page=wc-admin&path=/child' )
+			);
+			getHistory().push( getAdminLink( '/child' ) );
+
+			await new Promise( ( resolve ) => {
+				setTimeout( () => {
+					resolve();
+				}, 0 );
+			} );
```
Comment (psealock) [inline client/navigation/components/container/test/index.js:162]: Is this to wait for the event loop to finish?

**Review by psealock — APPROVED** [review]
Works well as described, tests look good too. Just a couple comments 🚢

File: TESTING-INSTRUCTIONS.md (hunk 1 above)
Comment (joshuatf) [reply TESTING-INSTRUCTIONS.md:11]: Good addition!  Added.

File: client/navigation/components/container/test/index.js (hunk 2 above)
Comment (joshuatf) [reply client/navigation/components/container/test/index.js:162]: Exactly.  We have a similar check in the history listener to correctly set the active item, so it also needs to be done here.

**Pull request merged**